homeassistant/components/ecobee/switch.py

```diff
@@ -0,0 +1,105 @@
+"""Support for using switch with ecobee thermostats."""
+
+from __future__ import annotations
+
+from collections.abc import Awaitable, Callable
+from dataclasses import dataclass
+from datetime import datetime
+import logging
+from typing import Any
+
+from homeassistant.components.switch import SwitchEntity
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+
+from . import EcobeeData
+from .const import DOMAIN
+from .entity import EcobeeBaseEntity
+
+_LOGGER = logging.getLogger(__name__)
+
+DATE_FORMAT = "%Y-%m-%d %H:%M:%S"
+
+
+@dataclass(frozen=True, kw_only=True)
+class EcobeeSwitchEntityDescription(SwitchEntity):
+    """Class describing Ecobee switch entities."""
+
+    ecobee_setting_key: str
+    set_fn: Callable[[EcobeeData, int, int], Awaitable]
+
+
+async def async_setup_entry(
+    hass: HomeAssistant,
+    config_entry: ConfigEntry,
+    async_add_entities: AddEntitiesCallback,
+) -> None:
+    """Set up the ecobee thermostat switch entity."""
+    data: EcobeeData = hass.data[DOMAIN]
+    _LOGGER.debug("Adding ventilators 20 min switch if present")
+
+    async_add_entities(
+        (
+            EcobeeVentilator20MinSwitch(data, index)
+            for index, thermostat in enumerate(data.ecobee.thermostats)
+            if thermostat["settings"]["ventilatorType"] != "none"
+        ),
+        True,
+    )
+
+
+class EcobeeVentilator20MinSwitch(EcobeeBaseEntity, SwitchEntity):
+    """A Switch class, representing 20 min timer for an ecobee thermostat with ventilator attached."""
+
+    def __init__(
+        self,
+        data: EcobeeData,
+        thermostat_index: int,
+    ) -> None:
+        """Initialize ecobee ventilator platform."""
+        super().__init__(data, thermostat_index)
+        self._attr_unique_id = "ventilator_20m_timer"
+        self._attr_native_value = False
+        self.update_without_throttle = False
+
+    @property
+    def is_on(self) -> bool:
+        """Get the latest state from the thermostat."""
+        return self._attr_native_value
+
+    async def async_update(self) -> None:
+        """Get the latest state from the thermostat."""
+
+        if self.update_without_throttle:
+            await self.data.update(no_throttle=True)
+            self.update_without_throttle = False
+        else:
+            await self.data.update()
+
+        ventilatorOffDateTime = self.thermostat["settings"]["ventilatorOffDateTime"]
+
+        time_zone_delay = datetime.strptime(
+            self.thermostat["utcTime"], DATE_FORMAT
+        ) - datetime.strptime(self.thermostat["thermostatTime"], DATE_FORMAT)
+
+        self._attr_native_value = (
+            ventilatorOffDateTime is not None
+            and ventilatorOffDateTime != ""
+            and datetime.strptime(ventilatorOffDateTime, DATE_FORMAT) + time_zone_delay
+            >= datetime.now()
```

Comment:
Either compare aware datetimes including timezone set or compare UTC times. Calculating the timezone delay and adding that to the local time without timezone seems like beating around the bush.

Please use our dt utils as much as possible, eg:

https://github.com/home-assistant/core/blob/822646749d32f346ffbb6c7ae5ebae06564fcfa0/homeassistant/util/dt.py#L107-L108

https://github.com/home-assistant/core/blob/822646749d32f346ffbb6c7ae5ebae06564fcfa0/homeassistant/util/dt.py#L103

Reply:
Ok to get the timezone of the device I have open a pr to the underlying lib to get access to that data. 
https://github.com/nkgilley/python-ecobee-api/pull/77

But this + the new code should do it. 

```python
def __init__(...)
...

        self._operating_timezone = dt_util.get_time_zone(
            self.thermostat["location"]["timeZone"]
        )

async def async_update(self) -> None:
...
dt_util.parse_datetime( ventilator_off_date_time, raise_on_error=True ).replace(tzinfo=self._operating_timezone)
            >= dt_util.now(self._operating_timezone)
```

Is that what you had in mind?

Reply:
Yes. Or maybe just use the timer UTC time since that seems to be available too in the library. Since we're not interested in the actual timestamp, we just want to compare times, we can compare the timer UTC timestamp with the current UTC time.

Reply:
We can get the UTC current time of the thermostat but not when the ventilator will stop.  

That value (stop time) is only available in the thermostat local timestamp, and this is the part I am trying to figure out.